byterun/caml/osdeps.h

```diff
@@ -91,12 +91,29 @@ extern char_os * caml_executable_name(void);
 */
 extern char_os *caml_secure_getenv(char_os const *var);
 
-/* Windows Unicode support */
+#ifdef _WIN32
+extern void caml_probe_os_version(void);
+#else
+#define caml_probe_os_version()
+#endif
+
+#if WINDOWS_UNICODE && defined(_WIN32)
+extern void caml_setup_terminal(void);
+#else
+#define caml_setup_terminal()
+#endif
 
 #ifdef _WIN32
 
 extern int caml_win32_rename(const wchar_t *, const wchar_t *);
 
+extern unsigned short caml_win32_major;
+extern unsigned short caml_win32_minor;
+extern unsigned short caml_win32_build;
+extern unsigned short caml_win32_revision;
+
```

Comment:
Presently, I don't think you're using these 4 variables outside of win32.c, so  the principle of encapsulation says they should be `static` in win32.c and not declared in osdeps.h.

However, it is entirely plausible that we will later have other uses of those variables elsewhere, in which case the declaration makes sense.

Reply:
In the earlier version, they were genuinely `extern` as they were referred to in sys.c, but I envisaged their being useful to have globally - in fact, it's a mistake that I'd put them inside a `CAML_INTERNALS` block by mistake. May I make them a global part of `osdeps.h` (guarded by #ifdef _WIN32 - although in principle these could be made available to the Cygwin port as well)